### PR TITLE
Admin WebApp

### PR DESCRIPTION
**Pull request by leorossi**

### CODE REVIEW

File: cognito.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+resource "aws_route53_record" "root_record" {
+  provider = aws.dns
+  name     = var.route53_zone
+  type     = "A"
+  zone_id  = data.aws_route53_zone.primary[0].id
+  records  = ["1.1.1.1"]
+  ttl      = 300
+}
+
+resource "aws_route53_record" "auth-cognito-A" {
```
Comment (dgonzalez) [inline cognito.tf]: Just for consistency, `_` for resource names instead of `-`

File: cognito.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  name         = "settings-read"
+  user_pool_id = aws_cognito_user_pool.admin_user_pool.id
+}
+
+resource "aws_cognito_user_group" "settings_write" {
+  name         = "settings-write"
+  user_pool_id = aws_cognito_user_pool.admin_user_pool.id
+}
+
+resource "aws_cognito_user_group" "otc-send" {
```
Comment (dgonzalez) [inline cognito.tf]: Same here and others as well (`-` vs `_`)

File: ecs_admin.tf [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    resources = [
+      data.aws_secretsmanager_secret_version.rds_read_only.arn,
+      data.aws_secretsmanager_secret_version.rds_read_write.arn,
+      data.aws_secretsmanager_secret_version.admin_push_service_token.arn,
+      data.aws_secretsmanager_secret_version.google_maps_api_key.arn
+    ]
+  }
+
+  statement {
+    actions = ["kms:*"]
```
Comment (dgonzalez) [inline ecs_admin.tf]: Is it possible to remove the `*` and target to the specific required actions?

File: ecs_admin.tf [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      "cognito-idp:AdminRemoveUserFromGroup",
+      "cognito-idp:ListUsers"
+    ]
+    resources = [
+      aws_cognito_user_pool.admin_user_pool.arn
+    ]
+  }
+
+  statement {
+    actions = ["sqs:*"]
```
Comment (dgonzalez) [inline ecs_admin.tf]: Same here.

File: gateway.tf [hunk 5]
```diff
@@ -158,6 +159,7 @@ resource "aws_api_gateway_integration_response" "isolation_get_integration_respo
     "method.response.header.Cache-Control"             = "'no-store'",
     "method.response.header.Pragma"                    = "'no-cache'",
     "method.response.header.Strict-Transport-Security" = format("'max-age=%s; includeSubDomains'", var.hsts_max_age)
+    "method.response.header.X-Frame-Options"           = "'SAMEORIGIN'"
```
Comment (dgonzalez) [inline gateway.tf]: Is this included in the Admin work? this looks like it potentially may break the NI notices of isolation as the IFRAME will be embedded in a different origin.

Comment (dgonzalez) [reply gateway.tf]: I think we will need to serve this from the backend if I am not wrong.

File: gateway.tf [hunk 6]
```diff
@@ -828,3 +1047,15 @@ resource "aws_api_gateway_authorizer" "main" {
...
+
+# #########################################
+# Default Responses
+# #########################################
+resource "aws_api_gateway_gateway_response" "test" {
+  rest_api_id   = aws_api_gateway_rest_api.main.id
+  response_type = "DEFAULT_4XX"
+
+  response_parameters = {
+    "gatewayresponse.header.access-control-allow-origin"  = "'*'"
```
Comment (dgonzalez) [inline gateway.tf]: we should try to remove the `*` in favour of a more restrictive policy.

File: secrets.tf [hunk 7]
```diff
@@ -25,6 +25,14 @@ data "aws_secretsmanager_secret_version" "rds" {
   secret_id = "${local.config_var_prefix}rds"
 }
 
+data "aws_secretsmanager_secret_version" "admin_push_service_token" {
```
Comment (dgonzalez) [inline secrets.tf:28]: This two secrets probably grant an update in the docs to include them (see the docs folder)

**Review by dgonzalez — CHANGES_REQUESTED** [review]

File: ecs_admin.tf (hunk 4 above)
Comment (leorossi) [reply ecs_admin.tf]: Removed as not used

File: ecs_admin.tf (hunk 3 above)
Comment (leorossi) [reply ecs_admin.tf]: Removed as not used

File: secrets.tf (hunk 7 above)
Comment (leorossi) [reply secrets.tf:28]: Docs added in a661406

File: gateway.tf (hunk 6 above)
Comment (leorossi) [reply gateway.tf]: This is only for default 4XX errors, I don't recall why we put that here but the `"test"` resource name suggests me we could remove entirely this resource maybe?

File: gateway.tf (hunk 5 above)
Comment (leorossi) [reply gateway.tf]: Removed, as should be sent correctly from ECS backend

**Pull request merged**